File: pgm/ABC_Java/ABC_428_A.java
import java.io.*;
// import java.math.*;
// import java.time.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        reader.close();
        final int one = 1;
        final int two = 2;
        final int three = 3;
        final int zero = 0;
        final int S = Integer.parseInt (arr[zero]);
        final int A = Integer.parseInt (arr[one]);
        final int B = Integer.parseInt (arr[two]);
        final int X = Integer.parseInt (arr[three]);
        System.out.println (S * A * (X / (A + B)) +
                            S * Math.min (A, X % (A + B)));
        System.exit (0);
    }
}

File: pgm/ABC_R/ABC_428_A.r
con <- file(description = "stdin", open = "r")
arr <- readLines(con = con)
close(con)
blank <- ""
space <- " "
yn <- "\n"
one <- 1L
two <- 2L
three <- 3L
four <- 4L
S <- strsplit(x = arr[one], split = space, fixed = TRUE)[[one]][one] |>
  as.integer()
A <- strsplit(x = arr[one], split = space, fixed = TRUE)[[one]][two] |>
  as.integer()
B <- strsplit(x = arr[one], split = space, fixed = TRUE)[[one]][three] |>
  as.integer()
X <- strsplit(x = arr[one], split = space, fixed = TRUE)[[one]][four] |>
  as.integer()
(S * A * (X %/% (A + B)) + S * min(A, X %% (A + B))) |>
  cat(yn, sep = blank)
q("no")